### ISSUE TITLE
How compare `&Path` and `&Ident` ?

### ISSUE THREAD

**Issue by davemilter** [dtolnay/syn#592]
I want compare `&syn::Path` and `&syn::Ident` 
there is method `Path::is_ident` for it: https://docs.rs/syn/0.15/syn/struct.Path.html ,
but I can not call it.

In other words I want to write something like this,
but such code doesn't compile.

```
fn f(p: &Path, i: &Ident) {
    p.is_ident(i);
}
```
I can fix it via call of `clone`, but creation of new object for comparing looks strange,
also I can introduce new type:
```
struct IdentRef<'a>(&'a Ident);

impl<'a> PartialEq<IdentRef<'_>> for Ident {
    fn eq(&self, o: &IdentRef<'_>) -> bool {
        self == o.0
    }
}
```
and fix `f` via `path.is_ident(IdentRef(id))`,
is it by design that is so hard to use or I missed something?

**Comment by dtolnay** [comment]
I think the signature of Path::is_ident is the one I want. You could try sending a PR to proc-macro2 adding whichever PartialEq impl is missing.

**Comment by dtolnay** [comment]
This is fixed in 1.0.0, you can now pass &Ident to path.is_ident.
